src/execution_tree/primitives/sum_operation.cpp

```diff
@@ -0,0 +1,290 @@
+// Copyright (c) 2018 Parsa Amini
+// Copyright (c) 2018 Hartmut Kaiser
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <phylanx/config.hpp>
+#include <phylanx/execution_tree/primitives/sum_operation.hpp>
+#include <phylanx/ir/node_data.hpp>
+#include <phylanx/util/matrix_iterators.hpp>
+
+#include <hpx/include/lcos.hpp>
+#include <hpx/include/naming.hpp>
+#include <hpx/include/util.hpp>
+#include <hpx/throw_exception.hpp>
+#include <hpx/util/iterator_facade.hpp>
+#include <hpx/util/optional.hpp>
+
+#include <algorithm>
+#include <cstddef>
+#include <cstdint>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <blaze/Math.h>
+
+///////////////////////////////////////////////////////////////////////////////
+namespace phylanx { namespace execution_tree { namespace primitives
+{
+    ///////////////////////////////////////////////////////////////////////////
+    primitive create_sum_operation(hpx::id_type const& locality,
+        std::vector<primitive_argument_type>&& operands,
+        std::string const& name, std::string const& codename)
+    {
+        static std::string type("sum_operation");
+        return create_primitive_component(
+            locality, type, std::move(operands), name, codename);
+    }
+
+    match_pattern_type const sum_operation::match_data =
+    {
+        hpx::util::make_tuple("sum_operation",
+        std::vector<std::string>{"sum_operation(_1, _2)"},
+        &create_sum_operation, &create_primitive<sum_operation>)
+    };
+
+    ///////////////////////////////////////////////////////////////////////////
+    sum_operation::sum_operation(
+        std::vector<primitive_argument_type>&& operands,
+        std::string const& name, std::string const& codename)
+      : primitive_component_base(std::move(operands), name, codename)
+    {}
+
+    primitive_argument_type sum_operation::sum0d(arg_type&& arg,
+        hpx::util::optional<std::int64_t> axis, bool keep_dims) const
+    {
+        if (axis && axis.value() != 0 && axis.value() != -1)
+        {
+            /*HPX_THROW_EXCEPTION(hpx::bad_parameter,
```

Comment:
If this is not needed, please remove the code.

Reply:
Fixed.